### ISSUE TITLE
Eliminate pause at end of music tracks?

### ISSUE THREAD

**Issue by peyre** [raceintospace/raceintospace#879]
I can't say how pleased I am that the corruption in the music (#209) has been fixed!  There is still a small discrepancy though.

Each track plays, then pauses for about 3 seconds before repeating.  In BARIS, when a track reaches its end, it repeats immediately.  In fact the transition is seamless - you don't even notice it.  Is it possible for us to replicate that?

This is totally not a big deal!  The corruption was, but this is merely a "would be nice" item.

**Comment by Alsed** [comment]
Hi Leon! 
I'm currently implementing the hq music files, but I've also edited them to eliminate the pauses, could you please clone this branch and test it?

`git clone --branch high_bitrate_audio --single-branch http://github.com/Alsed/raceintospace.git`

**Comment by peyre** [comment]
I'll get on that tomorrow morning!  Looking forward to it.

**Comment by peyre** [comment]
That seems to have done it!  It's a little hard to be sure, since music is kind of screwed up on my system since I upgraded to the latest Xubuntu.  But they're clearly running together without (or mostly without?) a pause in between.

**Comment by Alsed** [comment]
I remember you were having problems with the audio on your computer, I hope you can get that resolved.
Let me know if there are any tracks that you think need more trimming. I tried not only to eliminate the pauses but also to keep the rhythm right.

**Comment by peyre** [comment]
Thanks!  It's apparently a known issue in Ubuntu 24.04, so hopefully will get resolved before long.  It's annoying, and shows most prominently in RIS of all things (maybe the problem is worst when playing MIDIs?).

Spotting tracks that need trimming is one of those things I'm in a really poor position to judge, because of the audio issues on this system.  But maybe someone else can weigh in on this.

**Comment by peyre** [comment]
Now that that sound bug is fixed in Ubuntu (no more stuttering), I've gone through a few screens and yes, it looks like this is fixed!  Thanks, I'll close the one out.

**Comment by peyre** [comment]
The audio problems were resolved by an Ubuntu update, I believe.  I'd like to say btw that it's a real pleasure to hear the music playing correctly: tune plays, tune repeats immediately.  No pause in between, not partial other tracks, and no loud obnoxious buzzing.  Possibly the single biggest improvement in all of 2.0.
Incidentally I've added you to the credits as "ALSED".  If you'd like it to be by your name rather than your handle, just let me know how you'd like to be credited.  We're gearing up for a release, so now's the time to make such a change.
    On Sunday, August 11, 2024 at 08:20:33 AM PDT, alsed ***@***.***> wrote:  
 
 


I remember you were having problems with the audio on your computer, I hope you can get that resolved.
Let me know if there are any tracks that you think need more trimming. I tried not only to eliminate the pauses but also to keep the rhythm right.

**Comment by alsed** [comment]
Thank you Leon.
There's no need to put me in the credits, the job on the audio was fairly easy and most of my other code has been reverted anyways (and that's a bummer cause I really think it was more clear and easy to understand than the original).

Happy to see a new release is coming up, remember that the flatpack manifesto is also ready.

**Comment by hweimer** [comment]
@alsed, I totally agree with you that your changes to the event code were much more clearer than the original. But unfortunately, part of the clarity resulted from your code using concepts that were at odds with the game mechanics, creating several bugs (#908).

I probably should have @-mentioned you in the discussion so you could have joined in case you didn't see it. But then I didn't want to appear as finger-pointing because I consider your contributions to the game as highly valuable.

**Comment by alsed** [comment]
> [@alsed](https://github.com/alsed), I totally agree with you that your changes to the event code were much more clearer than the original. But unfortunately, part of the clarity resulted from your code using concepts that were at odds with the game mechanics, creating several bugs ([#908](https://github.com/raceintospace/raceintospace/issues/908)).
> 
> I probably should have @-mentioned you in the discussion so you could have joined in case you didn't see it. But then I didn't want to appear as finger-pointing because I consider your contributions to the game as highly valuable.

Thank you for your comments @hweimer ,
I redesigned all that part of the code, then I tested it, but only in the event news. It was fun and challenging to do, but It's true that  the variables I modified were required elsewhere, so I understand that the code wasn't working as expected and changes had to be made. Don't worry about it.

**Comment by alsed** [comment]
> alsed, it's very modest of you to say that. I'm still inclined to include you in the credits since you made a big contribution, but can leave you out if you think it's better you not be there.

Alright Leon, I'll send you and email with my name to be included in the credits, just so you don't put _"Alsed the Modest"_ or something else on it, lol.
